engine/src/main/scala/cromwell/engine/workflow/InitializeWorkflowBackendsActor.scala

```diff
@@ -0,0 +1,91 @@
+package cromwell.engine.workflow
+
+import akka.actor.{Props, ActorRef, LoggingFSM}
+import cromwell.backend.WorkflowBackendActor.{InitializationFailed, InitializationSuccess, Initialize}
+import cromwell.engine.workflow.InitializeWorkflowBackendsActor._
+
+object InitializeWorkflowBackendsActor {
+
+  /**
+    * States
+    */
+  sealed trait InitializeWorkflowBackendsActorState
+  case object Unstarted extends InitializeWorkflowBackendsActorState
+  case object InitializingBackends extends InitializeWorkflowBackendsActorState
+  case object Done extends InitializeWorkflowBackendsActorState
+
+  /**
+    * State data
+    */
+  case class InitializeWorkflowBackendsActorData(successes: Seq[ActorRef], failures: Map[ActorRef, Throwable])
+
+  /**
+    * Commands
+    */
+  sealed trait InitializeWorkflowBackendsActorCommand
+  case object StartInitializations extends InitializeWorkflowBackendsActorCommand
+
+  /**
+    * Responses
+    */
+  sealed trait InitializeWorkflowBackendsActorResponse
+  case object BackendInitializationsSucceeded extends InitializeWorkflowBackendsActorResponse
+  case class BackendInitializationsFailed(reasons: Map[ActorRef, Throwable]) extends InitializeWorkflowBackendsActorResponse
+
+  def props(workflowTag: String, backendsToInitialize: Seq[ActorRef]): Props = Props(new InitializeWorkflowBackendsActor(workflowTag, backendsToInitialize))
+
+}
+
+case class InitializeWorkflowBackendsActor(workflowTag: String, backendsToInitialize: Seq[ActorRef]) extends LoggingFSM[InitializeWorkflowBackendsActorState, InitializeWorkflowBackendsActorData] {
```

Comment:
Is this going to be refactored based on the new idea?.
Calling to BeforeAll and then Validate?